crux.el

```diff
@@ -99,6 +99,28 @@
   :type 'list
   :group 'crux)
 
+(defcustom crux-shell-func
```

Comment:
Keep in mind that shell and terminal are different things. I actually think it's better to have dedicated shell and term commands and configuration for both of them, as sometimes you need a shell, but sometimes only a terminal will do.

Reply:
That makes sense to me. So would you prefer `crux-shell-func`, `crux-term-func` and a corresponding (net new) `crux-visit-shell-buffer`?

Reply:
I have updated the PR with an implementation that distinguishes between shells and terminals